scroll-animations-1/Overview.bs

```diff
@@ -589,42 +615,178 @@ Computed value: As specified
 
-    The <<scroll-direction>>, if provided, determines the timeline's {{orientation}}.
+Note: Allowing animation-name to be used for selecting timeline enables most
+common animations to have to use a single name for both their keyframes and
+timeline which is simple and ergonomics. The 'animation-timeline' property gives
+additional control to authors to independently select keyframes and timeline if
+necessary.
 
-    The first <<scroll-offset>>, if provided, determines the timeline's {{startScrollOffset}}.
 
-    The second <<scroll-offset>>, if provided, determines the timeline's {{endScrollOffset}}.
+In this case, each possible value of type <<keyframes-name>> from
+'animation-name' has the following effects:
 
-    The <<time>> value, if specified, determines the timeline's {{timeRange}}.
+:   none
+::  No timelines and keyframes are specified at all, so there will be no
+    animation. Any other animations properties specified for this animation have no
+    effect.
+
+
+:   <<keyframes-name>>
+::  If ''@timeline'' rule with  the name specified by <<keyframes-name>>, then
+    the animation's [=timeline=] is a timeline whose property values are
+    provided by that rule. Otherwise the animation's [=timeline=] is a
+    {{DocumentTimeline}}, more specifically the <a>default document
+    timeline</a>.
+
+Note: Notice that the behavior for the case when no timeline with given name was
+found is different for either property. This ensures backward compatibility
+since most existing animations which have 'animation-name' specified expect to
+use <a>default document timeline</a>.
+
+
+## The '@timeline' at-rule ## {#timeline-at-rule}
+
+[=Scroll Timelines=] are specified in CSS using the <dfn>@timeline</dfn>
+at-rule, defined as follows:
+
+<pre>
+  @timeline = @timeline <<timeline-name>> { <<declaration-list>> }
+
+
+  <dfn>&lt;timeline-name></dfn> = <<custom-ident>> | <<string>>
+</pre>
+
+
+An ''@timeline'' rule has a name given by the <<custom-ident>> or <<string>> in
+its prelude. The two syntaxes are equivalent in functionality; the name is the
+value of the ident or string. As normal for <<custom-ident>>s and <<string>>s,
+the names are fully case-sensitive; two names are equal only if they are
+codepoint-by-codepoint equal. The <<custom-ident>> additionally excludes the
+none keyword.
+
+
+The <<declaration-list>> inside of ''@timeline'' rule can only contain the
+descriptors defined in this section.
+
+### the 'timeline-type' descriptor ### {#timeline-type-descriptors}
+
+
+<pre class='descdef'>
+  Name: timeline-type
+  For: @timeline
+  Value: scroll | time
+  Initial: time
+</pre>
 
-    The <<single-animation-fill-mode>> value, if specified, determines the timeline's {{fill}}.
+The 'timeline-type' determines the type of timeline and whose possible values
+have the following effects:
+
+:   time
+::  The animation's [=timeline=] is a {{DocumentTimeline}}, more specifically
+    the <a>default document timeline</a>.
```

Comment:
Any reason to limit to the default document timeline not any document timeline?

Reply:
This is the [behavior of the animation constructor](https://drafts.csswg.org/web-animations/#ref-for-dom-animation-animation) and I think we should match that.

Beside we have to choose  *a* timeline for auto. It is not clear to me why anything other than "default document timeline" makes sense.

Reply:
From what I read the only time based timeline in this spec allowed is the default document timeline. Web Animations API  allows instantiating document timelines and attaching animations to them.  Is it intentionally that we are limiting CSS animations to be attached to arbitrary document timeline?

Reply:
You are correct that timeline can be change in Web Animations API but not in CSS Animations.
I am not intending to change the time-based CSS Animation capabilities that in this patch thus the current wording.
But I believe the timeline syntax here is flexible enough to enable this and bring CSS Animation in parity with WebAnimation in this regard.

As for why the current limitation in CSS Animations, I can only guess:
  - There is very limited value to having a non-default timeline at the moment.
 - There was not CSS Syntax specified for timelines.

Given these two I can imagine why no one bothered allowing custom timelines in CSS Animations. This patch will rectify #2 but does not just add custom time-based timeline in CSS Animations.

Reply:
I will merge this PR as is. Happy to make modifications if my answer is not accurate or there is a better way to word this.